src/server/registry/manifest/manifest.go

```diff
@@ -17,37 +17,35 @@ package manifest
 import (
 	"github.com/goharbor/harbor/src/api/artifact"
 	"github.com/goharbor/harbor/src/api/repository"
-	"github.com/goharbor/harbor/src/common/utils"
 	"github.com/goharbor/harbor/src/common/utils/log"
 	"github.com/goharbor/harbor/src/internal"
-	ierror "github.com/goharbor/harbor/src/internal/error"
-	"github.com/goharbor/harbor/src/pkg/project"
 	"github.com/goharbor/harbor/src/server/registry/error"
 	"github.com/goharbor/harbor/src/server/router"
+	"github.com/gorilla/mux"
 	"github.com/opencontainers/go-digest"
 	"net/http"
 	"net/http/httputil"
+	"strings"
 )
 
 // NewHandler returns the handler to handler manifest requests
-// TODO the parameters aren't required, use the global variables instead
-func NewHandler(proMgr project.Manager, proxy *httputil.ReverseProxy) http.Handler {
+func NewHandler(proxy *httputil.ReverseProxy) http.Handler {
 	return &handler{
-		proMgr: proMgr,
-		proxy:  proxy,
+		repoCtl: repository.Ctl,
+		artCtl:  artifact.Ctl,
+		proxy:   proxy,
 	}
 }
 
 type handler struct {
-	proMgr project.Manager
-	proxy  *httputil.ReverseProxy
+	repoCtl repository.Controller
+	artCtl  artifact.Controller
+	proxy   *httputil.ReverseProxy
 }
 
 func (h *handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	switch req.Method {
-	case http.MethodHead:
-		h.head(w, req)
-	case http.MethodGet:
+	case http.MethodHead, http.MethodGet:
```

Comment:
Does it comply with the spec when you write the response body in HEAD?

Reply:
Will not write any data to the response body in HEAD as the request will be proxied to the backed docker registry. What the `get` method done is just check the existence of the artifact in database